WooCommerce/Classes/ViewRelated/Products/ProductSelector/TopProductsFromCachedOrdersProvider.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import Yosemite
+import Storage
+
+/// Encapsulates the ids of those products that were most or last sold among the cached orders
+///
+struct TopProductsFromCachedOrders: Equatable {
+    let popularProductsIds: [Int64]
+    let lastSoldProductsIds: [Int64]
+
+    static var empty: TopProductsFromCachedOrders {
+        TopProductsFromCachedOrders(popularProductsIds: [],
+                                    lastSoldProductsIds: [])
+    }
```

Comment:
Currently we're using `empty` just on tests, right? We could move this bit to an extension on the test class `TopProductsFromCachedOrdersProviderTests` instead:

```
// MARK: Test helper
//
extension TopProductsFromCachedOrders {
    static var empty: TopProductsFromCachedOrders {
        TopProductsFromCachedOrders(popularProductsIds: [],
                                    lastSoldProductsIds: [])
    }
}
```

Reply:
This will be used also on _production_ code on the next PR, so I prefer to leave it here so it can be easily applied.